### PR TITLE
Redistribute gaslift for groups with alq limit

### PR DESCRIPTION
**Pull request by totto82**
In master gaslift was only redistributed for group limited by rates

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/wells/GasLiftStage2.cpp [hunk 1]
```diff
@@ -938,11 +938,11 @@ addOrRemoveALQincrement(GradMap &grad_map, const std::string& well_name, bool ad
 
 bool
 GasLiftStage2::SurplusState::
-checkALQlimit(double delta_alq, double delta_gas)
+checkALQlimit()
 {
     if (this->max_glift) {
         double max_alq = *(this->max_glift);
-        if ((max_alq) < (this->alq + delta_alq)  ) {
+        if ((max_alq) < (this->alq)  ) {
```
Comment (hakonhagland) [inline opm/simulators/wells/GasLiftStage2.cpp:945]: It is not clear to me why this change will allow ALQ to be removed from groups limited by ALQ. Maybe add a comment here to clarify the reasoning?

Comment (totto82) [reply opm/simulators/wells/GasLiftStage2.cpp:945]: This is just a bug fix. For group control rates we would like to make sure that we dont remove so much gaslift that the  group rate can not be reached (hence the delta_) in the check. For ALQ the limits are strict and we shouldn't add a delta_alq.

Comment (hakonhagland) [reply opm/simulators/wells/GasLiftStage2.cpp:945]: Ok, I see, so it is not possible to remove too much ALQ in this case. Is this because the the increments will always be divisible by the group ALQ limit? What happens if a group has both ALQ and oil/gas/water limits? Can you add this as comment in the source code too? I think it could help us in the future to have some explanation.

Comment (hakonhagland) [reply opm/simulators/wells/GasLiftStage2.cpp:945]: Thanks!

**Comment by totto82** [comment]
jenkins build this please
